src/ln/channelmonitor.rs

```diff
@@ -2463,6 +2514,139 @@ impl ChannelMonitor {
+							return None;
+						}
+					// ...else just increase the previous feerate by 25% (because that's a nice number)
+					} else {
+						let fee = $old_feerate * $predicted_weight / 750;
+						if $amount <= fee {
+							log_trace!(self, "Can't 25% bump claiming on {}, amount {} is too small", $txid, $amount);
+							return None;
+						}
+						fee
+					};
+
+					let previous_fee = $old_feerate * $predicted_weight / 1000;
+					let min_relay_fee = $fee_estimator.get_min_relay_sat_per_1000_weight() * $predicted_weight / 1000;
+					// BIP 125 Opt-in Full Replace-by-Fee Signaling
+					// 	* 3. The replacement transaction pays an absolute fee of at least the sum paid by the original transactions.
+					//	* 4. The replacement transaction must also pay for its own bandwidth at or above the rate set by the node's minimum relay fee setting.
+					let new_fee = if new_fee < previous_fee + min_relay_fee {
+						new_fee + previous_fee + min_relay_fee - new_fee
+					} else {
+						new_fee
+					};
+					Some((new_fee, new_fee * 1000 / $predicted_weight))
+				}
+			}
+		}
+
+		let new_timer = Self::get_height_timer(height, cached_claim_datas.soonest_timelock);
+		let mut inputs_witnesses_weight = 0;
+		let mut amt = 0;
+		for per_outp_material in cached_claim_datas.per_input_material.values() {
+			match per_outp_material {
+				&InputMaterial::Revoked { ref script, ref is_htlc, ref amount, .. } => {
+					inputs_witnesses_weight += Self::get_witnesses_weight(if !is_htlc { &[InputDescriptors::RevokedOutput] } else if script.len() == OFFERED_HTLC_SCRIPT_WEIGHT { &[InputDescriptors::RevokedOfferedHTLC] } else if script.len() == ACCEPTED_HTLC_SCRIPT_WEIGHT { &[InputDescriptors::RevokedReceivedHTLC] } else { &[] });
+					amt += *amount;
+				},
+				&InputMaterial::RemoteHTLC { ref preimage, ref amount, .. } => {
+					inputs_witnesses_weight += Self::get_witnesses_weight(if preimage.is_some() { &[InputDescriptors::OfferedHTLC] } else { &[InputDescriptors::ReceivedHTLC] });
+					amt += *amount;
+				},
+				&InputMaterial::LocalHTLC { .. } => { return None; }
+			}
+		}
+		assert!(amt != 0);
+
+		let predicted_weight = bumped_tx.get_weight() + inputs_witnesses_weight;
+		let new_feerate;
+		if let Some((new_fee, feerate)) = RBF_bump!(amt, cached_claim_datas.feerate_previous, fee_estimator, predicted_weight as u64, txid) {
+			bumped_tx.output[0].value = amt - new_fee;
+			new_feerate = feerate;
+		} else {
+			return None;
+		}
+		assert!(new_feerate != 0);
+
+		for (i, (vout, per_outp_material)) in cached_claim_datas.per_input_material.iter().enumerate() {
+			match per_outp_material {
+				&InputMaterial::Revoked { ref script, ref pubkey, ref key, ref is_htlc, ref amount } => {
+					let sighash_parts = bip143::SighashComponents::new(&bumped_tx);
+					let sighash = hash_to_message!(&sighash_parts.sighash_all(&bumped_tx.input[i], &script, *amount)[..]);
```

Comment:
In the long term it would be super super nice if we could DRY up the transaction signing logic so that we use the same codepath to sign txn the first time and after an RBF. OK for now, but really sucks to have duplicate code that could fail differently :(

Reply:
Tracked in #400